docker/README.md

```diff
@@ -0,0 +1,105 @@
+Development with Docker
+---
+
+With `docker` and `docker-compose` you can easily spin up a development environment
+and start working on dendrite.
+
+### Requirements
+
+- docker
+- docker-compose (version 3+)
+
+### Configuration
+
+Copy the `dendrite-docker.yaml` file to the root of the project and rename it to
+`dendrite.yaml`. It already contains the defaults used in `docker-compose` for 
+networking so you will only have to change things like the `server_name` or to 
+toggle `naffka`.
+
+You can run the following `docker-compose` commands either from the top directory
+specifying the `docker-compose` file
+```
+docker-compose -f docker/docker-compose.yml <cmd>
+```
+or from within the `docker` directory 
+
+```
+docker-compose <cmd>
+```
+
+### Starting a monolith server
+
+For the monolith server you would need a postgres instance 
+
+```
+docker-compose up postgres
+```
+
+and the dendrite component from `bin/dendrite-monolith-server`
+
+```
+docker-compose up monolith
```

Comment:
Wouldn't it make more sense to split monolith and polylith into two files here? The advantage would be that we could just do `docker-compose -f docker/polylith.yml` instead of having all these commands you put below.

Reply:
The issue I encountered with running `docker-compose up` is that sometimes the ports won't be available immediately after the container is up (i.e postgres) so the other dependent components will fail.

For this to work we would need to add a script that performs this kind of check (e.g https://docs.docker.com/compose/startup-order/)

Reply:
We could probably add an option to the binaries to retry database connections on startup, which would solve this I think.

Reply:
retrying this is probably better anyway, because it will save a lot of trouble in automated deployment situations.